src/include/encoding.h

```diff
@@ -117,7 +117,18 @@ inline void decode(bool &v, bufferlist::const_iterator& p) {
     ceph_##etype e;							\
     ::ceph::decode_raw(e, p);						\
     v = e;								\
-  }
+  }									\
```

Comment:
Reviewer feedback requested:

I've only implemented a minimal set of encode/decode templates using two buffers which is sufficient for ObjectStore::Transaction and the messages MOSDRepOp, MOSDECSubWrite and MOSDECSubReadReply. Should the addition of further templates be left until they are required or should this pull request cover the other standard types in encoding.h and denc.h?

Reply:
Nah, I wouldn't add things until they are really necessary.
With this being said, I'm thinking on whether we truly need to (extensively) touch the dencoding framework.